src/realm/sync/noinst/protocol_codec.hpp

```diff
@@ -460,6 +461,26 @@ class ClientProtocol {
                                             received_changesets); // Throws
     }
 
+    static sync::ProtocolErrorInfo::Action string_to_action(const std::string& action_string)
+    {
+        using action = sync::ProtocolErrorInfo::Action;
+        static const std::unordered_map<std::string, action> mapping{
+            {"ProtocolViolation", action::ProtocolViolation},
+            {"ApplicationBug", action::ApplicationBug},
+            {"Warning", action::Warning},
+            {"Transient", action::Transient},
+            {"LogOut", action::LogOut},
```

Comment:
`LogOut` and `DeleteRealm` are 2 new actions we identified the client would need/benefit to receive from the server to cover all existing actions the client used to take. @jbreams If we agree on the names, @tkaye407 could add them to the server. (`LogOut` should be sent with  `ErrorBadAuthentication` and `DeleteRealm` with `ErrorPermissionDenied`).

Reply:
https://jira.mongodb.org/browse/BAAS-15269

Reply:
Names SGTM. They are mostly internal anyways

Reply:
thanks